thrall/app/lib/ThrallStreamProcessor.scala

```diff
@@ -0,0 +1,79 @@
+package lib
+
+import java.net.InetAddress
+import java.util.UUID
+
+import akka.Done
+import akka.actor.ActorSystem
+import akka.stream.Materializer
+import akka.stream.scaladsl.Source
+import com.amazonaws.services.kinesis.clientlibrary.lib.worker.{InitialPositionInStream, KinesisClientLibConfiguration}
+import com.contxt.kinesis.{KinesisRecord, KinesisSource}
+import lib.kinesis.ThrallEventConsumer
+import org.joda.time.DateTime
+import play.api.Logger
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+class ThrallStreamProcessor(config: ThrallConfig, consumer: ThrallEventConsumer, actorSystem: ActorSystem, materializer: Materializer) {
+
+  implicit val mat = materializer
+
+  private val workerId = InetAddress.getLocalHost.getCanonicalHostName + ":" + UUID.randomUUID()
+
+  private val kinesisConfig = {
+    import config.{thrallKinesisEndpoint, thrallKinesisDynamoEndpoint, awsRegion}
+    Logger.info(s"creating kinesis consumer with endpoint=$thrallKinesisEndpoint, region=$awsRegion")
+    kinesisClientLibConfig(
+      kinesisAppName = config.thrallKinesisStream,
+      streamName = config.thrallKinesisStream,
+      from = config.from
+    ).withKinesisEndpoint(thrallKinesisEndpoint)
+     .withDynamoDBEndpoint(thrallKinesisDynamoEndpoint)
+  }
+
+  val kinesisSource: Source[KinesisRecord, Future[Done]] = KinesisSource(kinesisConfig)
+
+  implicit val dispatcher = actorSystem.getDispatcher
+
+  def run(): Future[Done] = {
```

Comment:
Not quite sure I understand how we're returning a `Future[Done]` here.

Reply:
`Future[Done]` is essentially `Future[Unit]`. It is returned by akka streams - in this case from the `runForEach`.